### PR TITLE
DAG state channel http routes and block rumor handler

### PR DESCRIPTION
**Pull request by TheMMaciek**

### CODE REVIEW

File: modules/dag-l1/src/main/scala/org/tessellation/dag/l1/http/Routes.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+          case Right(hash) => Ok(hash.asJson)
+        }
+      } yield response
+  }
+
+  private val p2p: HttpRoutes[F] = HttpRoutes.of[F] {
+    case req @ POST -> Root / "consensus" / "data" =>
+      for {
+        peerBlockConsensusInput <- req.as[Signed[PeerBlockConsensusInput]]
+        _ <- Async[F].start(peerBlockConsensusInputQueue.offer(peerBlockConsensusInput))
```
Comment (marcinwadon) [inline modules/dag-l1/src/main/scala/org/tessellation/dag/l1/http/Routes.scala:39]: probably it should be a `Spawn[F]` instead of `Async[F]`

**Review by marcinwadon — CHANGES_REQUESTED** [review]

File: modules/dag-l1/src/main/scala/org/tessellation/dag/l1/infrastructure/block/rumor/handler.scala [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import org.tessellation.security.signature.Signed
+
+object handler {
+
+  def blockRumorHandler[F[_]: Async: KryoSerializer](
+    peerBlockQueue: Queue[F, Signed[DAGBlock]]
+  ): RumorHandler[F] =
+    RumorHandler.fromReceivedRumorFn[F, Signed[DAGBlock]]() {
+      case ReceivedRumor(_, signedBlock) =>
+        peerBlockQueue.offer(signedBlock)
```
Comment (marcinwadon) [inline modules/dag-l1/src/main/scala/org/tessellation/dag/l1/infrastructure/block/rumor/handler.scala:19]: I still suggest to make it through a storage/service instead directly on the queue :D

Comment (TheMMaciek) [reply modules/dag-l1/src/main/scala/org/tessellation/dag/l1/infrastructure/block/rumor/handler.scala:19]: If I make it through the storage I won't be able to stop the single stream with a semaphore and perform redownload e.g. That was my idea here.

Comment (TheMMaciek) [reply modules/dag-l1/src/main/scala/org/tessellation/dag/l1/infrastructure/block/rumor/handler.scala:19]: At least not as easily as I intended it to be :)

Comment (marcinwadon) [reply modules/dag-l1/src/main/scala/org/tessellation/dag/l1/infrastructure/block/rumor/handler.scala:19]: why? you shouldn't stop a queue itself but a service that pulls data

**Review by marcinwadon — APPROVED** [review]

**Pull request merged**